In [4]:
import numpy as np

The dataset is publicly available at https://www.kaggle.com/datasets/daviderusso7/seed-dataset

## Read the dataset files

In [19]:
dataset= np.load('SEED/DatasetCaricatoNoImage.npz')['arr_0']
labels= np.load('SEED/LabelsNoImage.npz')['arr_0']
subjects= np.load('SEED/SubjectsNoImage.npz')['arr_0']

## The dataset has the following shape: 

In [41]:
dataset.shape

(50910, 5, 62)

### This can be interpreted as (window , frequency band, channel). This means that for every window , we have its DE value for each frequency band across every channel.

The frequency bands used:

* Delta (1–4 Hz)

* Theta (4–8 Hz)

* Alpha (8–14 Hz)

* Beta (14–31 Hz)

* Gamma (31–50 Hz)

The labels are mapped like this 

0 → negative  
1 → neutral  
2 → positive

In [32]:
labels.shape,np.unique(labels)

((50910,), array([0, 1, 2]))

### The subject  ID is given for each sample

In [33]:
subjects.shape,np.unique(subjects)

((50910,),
 array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14],
       dtype=int32))

### Small EDA

In [42]:
print(f"Min feature value: {np.min(dataset)}")
print(f"Max feature value: {np.max(dataset)}")
print(f"Mean feature value: {np.mean(dataset)}")
print(f"Median feature value: {np.median(dataset)}")
print(f"Any negative values?: {np.any(dataset < 0)}")

Min feature value: 10.61709213256836
Max feature value: 42.113670349121094
Mean feature value: 19.94759750366211
Median feature value: 19.42363739013672
Any negative values?: False


## So for conclusion, the dataset has the following structure:

50910 samples

Each sample:

* 62 channels × 5 frequency-band DE values

* 1 label (emotion class)

* 1 subject ID